petisco/base/misc/wrapper.py

```diff
@@ -102,13 +112,7 @@ def wrapped(*args: Any, **kwargs: Any) -> Any:
                     )
                     logger.exception(exception)
 
-        try:
-            result.set_transformer(mapper.map)
-        except AttributeError:  # noqa
-            raise TypeError(
```

Comment:
this is now controlled previously as soon as possible to avoid repeat check in the new feature, also to ensure that middlewares receive a result object